components/server/src/bitbucket-server/bitbucket-server-repository-provider.ts

```diff
@@ -188,8 +188,21 @@ export class BitbucketServerRepositoryProvider implements RepositoryProvider {
         return commits.map((c) => c.id);
     }
 
-    // TODO: implement repo search
     public async searchRepos(user: User, searchString: string): Promise<RepositoryInfo[]> {
-        return [];
+        // Only load 1 page of 10 results for our searchString
+        const results = await this.api.getRepos(user, { maxPages: 1, limit: 10, searchString });
```

Comment:
Bitbucket Server search is limited to prefix searching only, so the name must match from the beginning. For example:

`test` would match `test-repo` but would not match `repo-test`